homeassistant/config_entries.py

```diff
@@ -1672,11 +1673,28 @@ async def async_finish_flow(
 
 
 class OptionsFlow(data_entry_flow.FlowHandler):
-    """Base class for config option flows."""
+    """Base class for config options flows."""
 
     handler: str
 
 
+class OptionsFlowWithConfigEntry(OptionsFlow):
+    """Base class for options flows with config entry."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+
+class OptionsFlowWithConfigEntryOptions(OptionsFlowWithConfigEntry):
```

Comment:
I created two classes because I'm not sure of the cost of `self.options = deepcopy(dict(config_entry.options))`

Since option flows are not initialised very frequently, I don't think cost would be a big issue so maybe we can use a single base class, and make `options` always available.